src/test/java/org/commcare/formplayer/tests/MenuAssertionsTest.java

```diff
@@ -0,0 +1,90 @@
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import org.commcare.formplayer.application.MenuController;
+import org.commcare.formplayer.beans.SessionNavigationBean;
+import org.commcare.formplayer.beans.menus.BaseResponseBean;
+import org.commcare.formplayer.beans.menus.CommandListResponseBean;
+import org.commcare.formplayer.beans.menus.EntityBean;
+import org.commcare.formplayer.beans.menus.EntityListResponse;
+import org.commcare.formplayer.configuration.CacheConfiguration;
+import org.commcare.formplayer.junit.InitializeStaticsExtension;
+import org.commcare.formplayer.junit.Installer;
+import org.commcare.formplayer.junit.RestoreFactoryExtension;
+import org.commcare.formplayer.junit.StorageFactoryExtension;
+import org.commcare.formplayer.junit.request.Response;
+import org.commcare.formplayer.junit.request.SessionNavigationRequest;
+import org.commcare.formplayer.services.RestoreFactory;
+import org.commcare.formplayer.utils.MockRequestUtils;
+import org.commcare.formplayer.utils.TestContext;
+import org.commcare.formplayer.web.client.WebClient;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+
+@WebMvcTest
+@Import({MenuController.class})
+@ContextConfiguration(classes = {TestContext.class, CacheConfiguration.class})
+@ExtendWith(InitializeStaticsExtension.class)
+public class MenuAssertionsTest {
+
+    private static final String ASSERTIONS_APP = "basic";
+    @Autowired
+    protected WebClient webClientMock;
+
+    @Autowired
+    RestoreFactory restoreFactoryMock;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    private MockRequestUtils mockRequest;
+
+    @RegisterExtension
+    static RestoreFactoryExtension restoreFactoryExt = new RestoreFactoryExtension.builder()
+            .withUser("user").withDomain("domain")
+            .withRestorePath("restores/basic.xml")
+            .build();
+
+    @RegisterExtension
+    static StorageFactoryExtension storageExt = new StorageFactoryExtension.builder()
+            .withUser("user").withDomain("domain").build();
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        mockRequest = new MockRequestUtils(webClientMock, restoreFactoryMock);
+    }
+
+
+    @Test
+    public void testMenuAssertions() {
+        // Check passing assertion
+        Response<CommandListResponseBean> response0 = navigate(new String[]{"0"},
+                CommandListResponseBean.class);
+        assertNotNull(response0);
+        // Check failing assertion
+        assertThrows(Exception.class, () -> {
```

Comment:
can we also verify that exception has the right message as per assertions here.

Reply:
verified here https://github.com/dimagi/formplayer/pull/1411/commits/19a982c29550c2e8bbc520bca803cbf84eb764a2